logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java

```diff
@@ -0,0 +1,27 @@
+package com.logstash;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+public class KeyNodeTest {
```

Comment:
Well I think this is effectively a bug in our logic.

lets take the ruby implementation of the following:
```ruby 
["foo", nil, "bar"].join(",")
=> "foor,,bar"
```